packages/legend-query/src/stores/filterOperators/QueryBuilderFilterOperator_Equal.ts

```diff
@@ -84,22 +84,20 @@ export class QueryBuilderFilterOperator_Equal extends QueryBuilderFilterOperator
     const type = filterConditionState.value
       ? getNonCollectionValueSpecificationType(filterConditionState.value)
       : undefined;
+
+    const numericPrimitiveTypes = [
+      PRIMITIVE_TYPE.NUMBER,
+      PRIMITIVE_TYPE.INTEGER,
+      PRIMITIVE_TYPE.DECIMAL,
+      PRIMITIVE_TYPE.FLOAT,
+    ] as string[];
+
     return (
       type !== undefined &&
-      ((
-        [
-          PRIMITIVE_TYPE.STRING,
-          PRIMITIVE_TYPE.BOOLEAN,
-          PRIMITIVE_TYPE.NUMBER,
-          PRIMITIVE_TYPE.INTEGER,
-          PRIMITIVE_TYPE.DECIMAL,
-          PRIMITIVE_TYPE.FLOAT,
-          PRIMITIVE_TYPE.DATE,
-          PRIMITIVE_TYPE.STRICTDATE,
-          PRIMITIVE_TYPE.DATETIME,
-        ] as string[]
-      ).includes(type.path) ||
-        type === propertyType)
+      ((numericPrimitiveTypes.includes(type.path) &&
```

Comment:
I thought we can use `isSuperType` for number types as well?

Reply:
Also, what about other operators? Why only `equal`?

Reply:
For number, i don't think we could because we could assign integer to float, etc.

Reply:
you mean the LHS can be narrower than the RHS, i.e. `someIntegerProperty == 123.123`? Did you execute to confirm this?

Reply:
Like said, let's check. I don't think the following could work:

```
LHS (Integer) == RHS (Float)
```

Can you check?

Reply:
Ok, so this is not straight-forward behavior at all. Let's document this in a comment here about why we do this.